### DISCUSSION TITLE
Is creating model grid from elevation data possible?

### DISCUSSION THREAD

**Discussion by AH-CSM** [modflowpy/flopy#1453] [Q&A] [answered]
Is it possible to import a gis shape file or csv elevation file, or some other data file to be filtered into the disu or disv package? I would like to create a unique model grid based on the specific subsurface geology, but it seems time-consuming and challenging to identify individual cells to be excluded with an idomain command if there are numerous layers and differences in elevation between layers.
Thank you.

**Comment by jlarsen-usgs** [comment, marked as answer]
@AH-CSM 

We have support to create model grid elevations and even idomain/ibound arrays from raster data. Take a look at [The raster intersection notebook](https://github.com/modflowpy/flopy/blob/develop/examples/Notebooks/flopy3_raster_intersection.ipynb).

One requirement for this, is that you will need to create a minimal modelgrid prior to doing the intersections. Your modelgrid cells in at least 2d must be represented in geospatial coordinates prior to intersecting with raster and shapefile data.

Example code.

```
from flopy.discretization import StructuredGrid
from flopy.utils import Raster

delr = np.ones((10,)) * 250
delc = np.ones((10,)) * 250 
xll = 10
yll = 100
modelgrid = StructuredGrid(delc=delc, delr=delr, xoff=xll, yoff=yll)

rstr = Raster.load("my raster")
resampled = rstr.resample_to_grid(
    modelgrid,
    band=rstr.bands[0],
    method="nearest",
)
